### DISCUSSION TITLE
UNSCALE DATA

### DISCUSSION THREAD

**Discussion by daryraa** [teamalgoritma/community#306] [Q&A] [answered]
Permisi kak cara supaya kita bisa mengunscale suatu data gimana ya soalnya saya agak bingung membaca angkanya 
![image](https://user-images.githubusercontent.com/78478403/190577549-985a6b02-5e83-4490-b981-5ef549b941fb.png)
kalau kita ngescale kan menggunakan fungsi scale() apakah untuk meng unscale ada caranya?

**Comment by NabiilahArdini** [comment, marked as answer]
Halo @daryraa ,
untuk melakukan unscaled, dapat dilakukan operasi matematika yang mengembalikan nilainya ke bentuk asli, yakni mengalikan nilai dengan standar deviasi kolom awal, kemudian menambahkannya dengan mean kolom awal. Sayangnya, tidak terdapat fungsi bawaan di R untuk unscale data. Oleh karena itu perlu dibuat fungsi manual.

Berikut fungsi yang coba saya buat:
```
# buat fungsi unscale
unscale <- function(x){
  
  m <- attr(x, "scaled:center")
  sd <- attr(x, "scaled:scale")
  
  dat <- (t(x) * sd) + m 
  as.data.frame(t(dat))
  
}
```
Contoh penggunaan:
```
dat <- data.frame(usia = c(23,24,25,22),
                  pendapatan = c(3,3,3.5,4))

dat_s <- scale(dat)
```
<img width="190" alt="image" src="https://user-images.githubusercontent.com/51941102/190946325-ba45b543-7d5f-4d00-8606-18e30cdf9f82.png">

```
unscale(dat_s)
```
<img width="270" alt="image" src="https://user-images.githubusercontent.com/51941102/190946350-6cf7a2c9-f002-4ed8-adc5-d371967d9cb9.png">

Silahkan mencoba. 

Sebagai note tambahan, data yang dimasukan ke fungsi unscale harus hanya berupa data yg dimasukan ke fungsi scale, sehingga kolom cluster tidak bisa diikutsertakan. Saran saya, Anda bisa memasukan hasil clustering ke data yang belum discale saja, kemudian melakukan profiling dari sana.

Apabila jawaban ini membantu, boleh berikan "Mark as Answer". 
Semoga informasi ini bermanfaat. 

Best regards!

**Reply by daryraa** [reply]
baik kak terima kasih
akan saya coba
